azkaban-execserver/src/main/java/azkaban/execapp/jmx/JmxJobMBeanManager.java

```diff
@@ -0,0 +1,164 @@
+package azkaban.execapp.jmx;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.log4j.Logger;
+
+import azkaban.event.Event;
+import azkaban.event.EventListener;
+import azkaban.execapp.JobRunner;
+import azkaban.executor.ExecutableNode;
+import azkaban.executor.Status;
+import azkaban.utils.Props;
+
+/**
+ * Responsible keeping track of job related MBean attributes through listening
+ * to job related events.
+ * 
+ * @author hluu
+ *
+ */
+public class JmxJobMBeanManager implements JmxJobMXBean, EventListener {
+
+  private static final Logger logger = Logger
+      .getLogger(JmxJobMBeanManager.class);
+
+  private static JmxJobMBeanManager INSTANCE = new JmxJobMBeanManager();
+
+  private AtomicInteger runningJobCount = new AtomicInteger(0);
+  private AtomicInteger totalExecutedJobCount = new AtomicInteger(0);
+  private AtomicInteger totalFailedJobCount = new AtomicInteger(0);
+  private AtomicInteger totalSucceededJobCount = new AtomicInteger(0);
+
+  private Map<String, AtomicInteger> jobTypeFailureMap =
+      new HashMap<String, AtomicInteger>();
+
+  private Map<String, AtomicInteger> jobTypeSucceededMap =
```

Comment:
True.  However synchronization is already done in method handleJobFinishedByType.